packages/aila/src/core/chat/AilaStreamHandler.ts

```diff
@@ -23,31 +21,49 @@ export class AilaStreamHandler {
 
   public startStreaming(abortController?: AbortController): ReadableStream {
     return new ReadableStream({
-      start: async (controller) => {
-        await this.stream(controller, abortController);
+      start: (controller) => {
+        this.stream(controller, abortController).catch((error) => {
+          log.error("Error in stream:", error);
+          controller.error(error);
+        });
       },
     });
   }
 
+  private logStreamingStep(step: string) {
+    log.info(`Streaming step: ${step}`);
+  }
+
   private async stream(
     controller: ReadableStreamDefaultController,
     abortController?: AbortController,
   ) {
     this.setupController(controller);
-    this.listenForAbort(abortController);
     try {
       await this._chat.setupGeneration();
+      this.logStreamingStep("Setup generation complete");
```

Comment:
In the absence of tracing, these help understand what's happening during streaming and how long each step is taking